src/Features/Core/Portable/DocumentationCommentFormatting/AbstractDocumentationCommentFormattingService.cs

```diff
@@ -203,44 +182,36 @@ private static void AppendTextFromNode(FormatterState state, XNode node, Compila
             }
         }
 
-        private static void AppendTextFromTagWithNameAttribute(FormatterState state, XElement element, Compilation compilation)
+        private static void AppendTextFromAttribute(FormatterState state, XElement element, string attributeName)
         {
-            var nameAttribute = element.Attribute("name");
-
-            if (nameAttribute == null)
+            var attribute = element.Attribute(attributeName);
+            if (attribute == null)
             {
                 return;
             }
 
-            if (compilation != null && state.TryAppendSymbol(DocumentationCommentId.GetFirstSymbolForDeclarationId(nameAttribute.Value, compilation)))
-            {
-                return;
-            }
-            else
-            {
-                state.AppendString(TrimCrefPrefix(nameAttribute.Value));
-            }
+            state.AppendParts(CrefToSymbolDisplayParts(attribute.Value, state.Position, state.SemanticModel, state.Format));
         }
 
-        private static void AppendTextFromSeeTag(FormatterState state, XElement element, Compilation compilation)
+        internal static IEnumerable<SymbolDisplayPart> CrefToSymbolDisplayParts(string crefValue, int position, SemanticModel semanticModel, SymbolDisplayFormat format = null)
         {
-            var crefAttribute = element.Attribute("cref");
-
-            if (crefAttribute == null)
+            // first try to parse the symbol
+            if (semanticModel?.Compilation != null)
```

Comment:
The `FormatterState.TryAppendSymbol()` method that I removed explicitly checked for a null semantic model so I simply carried that check forward.  As for `Compilation`, I have no idea if that can be null or not, but if it is `DocumentationCommentId.GetFirstSymbolForDeclarationId()` will throw if the compilation was null and I was trying to avoid that, but if a semantic model's compilation can never be null (or if it is then something else catastrophic happened) then I'm ok to turn

```C#
if (semanticModel?.Compilation != null)
```

into

``` C#
if (semanticModel != null) // if the model is valid, then so is the compilation
```